lib/finish_unfinished_persons.rb

```diff
@@ -108,8 +109,13 @@ def self.compute_semi_id(competition_year, person_name, available_per_semi = {})
     sanitized_roman_name = self.remove_accents roman_name
     name_parts = sanitized_roman_name.gsub(/[^a-zA-Z ]/, '').upcase.split
 
-    last_name = name_parts[-1]
-    rest_of_name = name_parts[...-1].join
+    if name_parts.length > 1 && GENERATIONAL_SUFFIXES.include?(name_parts[-1])
+      last_name = name_parts[-2]
+      rest_of_name = name_parts[0...-2].join
+    else
+      last_name = name_parts[-1]
+      rest_of_name = name_parts[0...-1].join
+    end
```

Comment:
The two if-branches look almost identical. I think what you're trying to do is compute an index which represents the "proper" family name, and then slice on that index. So I think you should do exactly that in code: Determine an integer (which defaults to `-1` and if the generational suffix check is true, gets set to `-2`) and then just slice once based on that index.